service/storage_service.cc

```diff
@@ -4964,6 +4964,40 @@ future<> storage_service::do_clusterwide_vnodes_cleanup() {
     rtlogger.info("cluster-wide vnodes cleanup done");
 }
 
+future<> storage_service::reset_cleanup_needed() {
+    auto& server = _group0->group0_server();
+    auto holder = _group0->hold_group0_gate();
+
+    while (true) {
+        auto guard = co_await _group0->client().start_operation(_group0_as, raft_timeout{});
+
+        auto me = _topology_state_machine._topology.find(server.id());
+        if (!me) {
```

Comment:
I see that in `start_vnodes_cleanup_on_dirty_nodes` we consider only normal nodes, but here we allow the node to be new/transitioning. I don't see any problem with this, but I'm also not sure that there is no problem.

Reply:
I do not see any problem either, but will restrict to normal just in case.